11_FFTBloom/app_resources/image_ifft_first_axis.hlsl

```diff
@@ -0,0 +1,164 @@
+#include "common.hlsl"
+
+/*
+ * Remember we have these defines:
+ * _NBL_HLSL_WORKGROUP_SIZE_
+ * ELEMENTS_PER_THREAD
+ * KERNEL_SCALE
+ * scalar_t
+ * FORMAT
+*/
+
+
+#define FFT_LENGTH (_NBL_HLSL_WORKGROUP_SIZE_ * ELEMENTS_PER_THREAD)
+
+[[vk::binding(0, 0)]] [[vk::image_format( FORMAT )]] RWTexture2D<float32_t4> convolvedImage;
+
+groupshared uint32_t sharedmem[workgroup::fft::SharedMemoryDWORDs<scalar_t, _NBL_HLSL_WORKGROUP_SIZE_>];
+
+// Users MUST define this method for FFT to work
+uint32_t3 glsl::gl_WorkGroupSize() { return uint32_t3(_NBL_HLSL_WORKGROUP_SIZE_, 1, 1); }
+
+struct SharedMemoryAccessor
+{
+	void set(uint32_t idx, uint32_t value)
+	{
+		sharedmem[idx] = value;
+	}
+
+	void get(uint32_t idx, NBL_REF_ARG(uint32_t) value)
+	{
+		value = sharedmem[idx];
+	}
+
+	void workgroupExecutionAndMemoryBarrier()
+	{
+		glsl::barrier();
+	}
+
+};
```

Comment:
1. consteval param struct / no macros
2. shared mem and accessor in common header?
3. don't spec the storage image format (update DXC sumodule to the master one)

Reply:
All steps but the normalization require sharedmem, and requiring it in normalization kills occupancy. So either:
- Keep in common, add a `#define` flag that optionally disables sharedmem if you `#define` it before including common (so normalization step defines this flag)
- Make another .hlsl that all shaders but normalization include

I'm inclined to do the first but maybe that's a bit ugly

Reply:
> Make another .hlsl that all shaders but normalization include

make a `fft_common` for all ffft shaders, thats the goto solution